ISSUER_PROTOCOL.md

```diff
@@ -0,0 +1,343 @@
+
+
+## Trust Token Crypto Protocol
+
+The Trust Token crypto protocol is based on the PMBTokens design in https://eprint.iacr.org/2020/072 (appendix H) using P-384. The necessary keys and function mappings are described below.
+
+
+### Keys
+
+The Trust Token protocol primarily requires two kinds of keypairs:
+
+*   SRRSigningKey/SRRVerificationKey - A Ed25519 public keypair for signing and verifying the integrity of the Signed Redemption Record response.
+*   TrustTokenSecretKey/TrustTokenPublicKey - A Trust Token keypair used to sign and verify Trust Tokens.
+
+#### SRRSigningKey/SRRVerificationKey
+
+These keys should be generated and stored as standard Ed25519 keys, with the public/verification key being included in the Key Commitment result as a 32-byte RFC8032 encoding.
+
+
+#### TrustTokenSecretKey/TrustTokenPublicKey
+
+These are keys used in Trust Token consisting of elliptic curve scalars and points. All scalars and points are sized based on the curve choice (P-384). Up to 3 keys may be configured in parallel in the key commitment.
+
+Each keypair consist of the following:
+
+
+```
+struct {
+  // Corresponding to the FALSE private metadata bit (x0, y0).
+  Scalar x0;
+  Scalar y0;
+  // Corresponding to the TRUE private metadata bit (x1, y1).
+  Scalar x1;
+  Scalar y1;
+  // Corresponding to the token validity (x˜, y˜).
+  Scalar xs;
+  Scalar ys;
+} TrustTokenSecretKey;
+
+struct {
+  ECPoint pub0; // Corresponding to the FALSE private metadata bit.
+  ECPoint pub1; // Corresponding to the TRUE private metadata bit.
+  ECPoint pubs; // Corresponding to the token validity check.
+} TrustTokenPublicKey;
+```
+
+The TrustTokenPublicKey is encoded in the key commitment and should be serialized as follows:
+
+
+```
+struct {
+  opaque pub0<1..2^16-1>; // X9.62 Uncompressed point.
+  opaque pub1<1..2^16-1>; // X9.62 Uncompressed point.
+  opaque pubs<1..2^16-1>; // X9.62 Uncompressed point.
+} TrustTokenPublicKey;
+```
+
+### Serialization/Hashing
+
+For the PMBTokens functions, the following serialization schemes and hashes are used internally using draft 07 of the hash-to-curve specification (https://tools.ietf.org/html/draft-irtf-cfrg-hash-to-curve-07):
+
+
+```
+Ht(t) - P384_XMD:SHA-512_SSWU_RO_ with a big-endian bytestring input of 't' and a dst of "PMBTokens Experiment V1 HashT".
```

Comment:
Having these lines inside code blocks means they won't be line-wrapped if they're really long. (And they kinda are. :-) ) Maybe they should be in text blocks like:

`Ht(t)` is defined to be P384_XMD:SHA-512_SSWU_RO_ with a big-endian bytestring input of `t` and a dst of "PMBTokens Experiment V1 HashT".

Reply:
Done.